sql: TestStreamerTightBudget failed

sql.TestStreamerTightBudget [failed](https://tanzanite.cluster.engflow.com/invocations/default/fe2f43e7-4d0f-4d99-8602-fa507e042fb9?testReportRun=17&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsOnNxbF90ZXN0) on master @ [04f0416d526a43741d22fd03966758dcccdeb79f](https://github.com/cockroachdb/cockroach/commits/04f0416d526a43741d22fd03966758dcccdeb79f):


```
=== RUN   TestStreamerTightBudget
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestStreamerTightBudget1831396537
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:144: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    mem_limit_test.go:207: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/sql/mem_limit_test.go:207
        	Error:      	"2.1" is not greater than or equal to "5.3"
        	Test:       	TestStreamerTightBudget
        	Messages:   	unexpectedly high memory usage
    panic.go:523: -- test log scope end --
test logs left over in: outputs.zip/logTestStreamerTightBudget1831396537
--- FAIL: TestStreamerTightBudget (0.43s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=17</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStreamerTightBudget.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/mem_limit_test.go
+++ b/pkg/sql/mem_limit_test.go
@@ -23,6 +23,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/sem/eval"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
+	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/errors"
@@ -161,42 +162,39 @@ func TestStreamerTightBudget(t *testing.T) {
 	s, db, _ := serverutils.StartServer(t, base.TestServerArgs{
 		SQLMemoryPoolSize: 1 << 30, /* 1GiB */
 	})
-	ctx := context.Background()
-	defer s.Stopper().Stop(ctx)
+	defer s.Stopper().Stop(context.Background())
+	sqlDB := sqlutils.MakeSQLRunner(db)
 
 	const blobSize = 1 << 20
 	const numRows = 5
 
-	_, err := db.Exec("CREATE TABLE t (pk INT PRIMARY KEY, k INT, blob STRING, INDEX (k))")
-	require.NoError(t, err)
+	sqlDB.Exec(t, "CREATE TABLE t (pk INT PRIMARY KEY, k INT, blob STRING, INDEX (k))")
 	for i := 0; i < numRows; i++ {
 		if i > 0 {
 			// Create a new range for this row.
-			_, err = db.Exec(fmt.Sprintf("ALTER TABLE t SPLIT AT VALUES(%d)", i))
-			require.NoError(t, err)
+			sqlDB.Exec(t, fmt.Sprintf("ALTER TABLE t SPLIT AT VALUES(%d)", i))
 		}
-		_, err = db.Exec(fmt.Sprintf("INSERT INTO t SELECT %d, 1, repeat('a', %d)", i, blobSize))
-		require.NoError(t, err)
+		sqlDB.Exec(t, fmt.Sprintf("INSERT INTO t SELECT %d, 1, repeat('a', %d)", i, blobSize))
 	}
 
 	// Populate the range cache.
-	_, err = db.Exec("SELECT count(*) from t")
-	require.NoError(t, err)
+	sqlDB.Exec(t, "SELECT count(*) from t")
 
 	// Set the workmem limit to a low value such that it will allow the Streamer
 	// to have at most one request to be "in progress".
-	_, err = db.Exec(fmt.Sprintf("SET distsql_workmem = '%dB'", blobSize))
-	require.NoError(t, err)
+	sqlDB.Exec(t, fmt.Sprintf("SET distsql_workmem = '%dB'", blobSize))
 
 	// Perform an index join to read the blobs.
 	query := "EXPLAIN ANALYZE SELECT sum(length(blob)) FROM t@t_k_idx WHERE k = 1"
 	maximumMemoryUsageRegex := regexp.MustCompile(`maximum memory usage: (\d+\.\d+) MiB`)
-	rows, err := db.QueryContext(ctx, query)
-	require.NoError(t, err)
-	for rows.Next() {
-		var res string
-		require.NoError(t, rows.Scan(&res))
-		if matches := maximumMemoryUsageRegex.FindStringSubmatch(res); len(matches) > 0 {
+	rows := sqlDB.QueryStr(t, query)
+	// Build pretty output for an error message in case we need it.
+	var sb strings.Builder
+	for _, row := range rows {
+		sb.WriteString(row[0] + "\n")
+	}
+	for _, row := range rows {
+		if matches := maximumMemoryUsageRegex.FindStringSubmatch(row[0]); len(matches) > 0 {
 			usage, err := strconv.ParseFloat(matches[1], 64)
 			require.NoError(t, err)
 			// We expect that the maximum memory usage is about 2MiB (1MiB is
@@ -204,7 +202,7 @@ func TestStreamerTightBudget(t *testing.T) {
 			// by the ColIndexJoin when the blob is copied into the columnar
 			// batch). We allow for 0.1MiB for other memory usage in the query.
 			maxAllowed := 2.1
-			require.GreaterOrEqual(t, maxAllowed, usage, "unexpectedly high memory usage")
+			require.GreaterOrEqualf(t, maxAllowed, usage, "unexpectedly high memory usage\n----\n%s", sb.String())
 			return
 		}
 	}

